### PR TITLE
Api data

### PR DESCRIPTION
**Pull request by mvigneron546**
This PR accomplishes the following points:

- Saves each request and response to and from the API to our Azure blob storage. 
- Saves each version of the train and test files to the blob storage
- Saves our API credentials (unchanged) to a secure location on the cloud, namely the Azure Key Vault
- Sets up a schedule job for the automatic data collection (so far every 30 mins, for testing purposes, to be changed later to once a week)

The docker-compose file has been accordingly enriched, as well as the data_collection script and the main.py file for the api service. The docker containers can be built as intended and works as well as before. Also a .env file has been written to centralize the location of environment variables, which are then made available for the docker builds.

WARNING: you will need to create a .env file containing the following environment variable:

- STORAGE_ACCOUNT_CONNECTION_STRING
- AZURE_CLIENT_ID
- AZURE_CLIENT_SECRET
- AZURE_TENANT_ID

I will not write the values of the variables here for security reasons, I will write them to you per email/chat.

Let me know if anything breaks!

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -1,17 +1,20 @@
...
+        context: ./api
+      ports:
+        - "8000:8000"
+      volumes:
+        - ./data:/app/data
+        - ./config.yml:/app/config.yml:ro
+      env_file:
+        - .env
+      depends_on:
+        - modeling
```
Comment (dbursy) [inline docker-compose.yml:20]: Replace with
modeling:
[ADD INDENTATION HERE] condition: service_completed_successfully

**Review by dbursy — APPROVED** [review]
Comment added

File: docker-compose.yml (hunk 1 above)
Comment (mvigneron546) [reply docker-compose.yml]: Good point - added this to docker-compose.yml
Thanks!

**Comment by mvigneron546** [comment]
Thanks for reviews @kevorks and @dbursy - merging now!

**Pull request merged**